src/ir_Mitsubishi.cpp

```diff
@@ -1186,3 +1198,449 @@ String IRMitsubishi136::toString(void) {
+  }
+
+  if (data[kMitsubishi112StateLength-1] != checksum) return false;
+
+  return true;
+}
+
+// Configure the pin for output.
+void IRMitsubishi112::begin(void) { _irsend.begin(); }
+
+#if SEND_MITSUBISHI112
+// Send the current desired state to the IR LED.
+void IRMitsubishi112::send(const uint16_t repeat) {
+  checksum();  // Ensure correct checksum before sending.
+  _irsend.sendMitsubishi112(remote_state, kMitsubishi112StateLength, repeat);
+}
+#endif  // SEND_MITSUBISHI112
+
+// Return a pointer to the internal state date of the remote.
+uint8_t *IRMitsubishi112::getRaw(void) {
+  checksum();
+  return remote_state;
+}
+
+void IRMitsubishi112::setRaw(const uint8_t *data) {
+  for (uint8_t i = 0; i < (kMitsubishi112StateLength - 1); i++) {
+    remote_state[i] = data[i];
+  }
+  checksum();
+}
+
+// Set the requested power state of the A/C to off.
+void IRMitsubishi112::on(void) { setPower(true); }
+
+// Set the requested power state of the A/C to off.
+void IRMitsubishi112::off(void) { setPower(false); }
+
+// Set the requested power state of the A/C.
+void IRMitsubishi112::setPower(bool on) {
+  // FIXME - Hardcoded values rather than anything else at the moment.
+  if (on)
+    remote_state[kMitsubishi112PowerByte] = 0x24;
+  else
+    remote_state[kMitsubishi112PowerByte] = 0x20;
+}
+
+// Return the requested power state of the A/C.
+bool IRMitsubishi112::getPower(void) {
+  return remote_state[kMitsubishi112PowerByte] & kMitsubishi112PowerBit;
+}
+
+// Set the temp. in deg C
+void IRMitsubishi112::setTemp(const uint8_t degrees) {
+  uint8_t temp = std::max((uint8_t)kMitsubishi112MinTemp, degrees);
+  temp = std::min((uint8_t)kMitsubishi112MaxTemp, temp);
+  remote_state[kMitsubishi112TempByte] = kMitsubishiAcMaxTemp - temp;
+}
+
+// Return the set temp. in deg C
+uint8_t IRMitsubishi112::getTemp(void) {
+  return (kMitsubishiAcMaxTemp -remote_state[kMitsubishi112TempByte]);
+}
+
+void IRMitsubishi112::setFan(const uint8_t speed) {
+  switch (speed) {
+    case kMitsubishi112FanMin:
+    case kMitsubishi112FanLow:
+    case kMitsubishi112FanMed:
+    case kMitsubishi112FanMax:
+      remote_state[kMitsubishi112FanByte] &= kMitsubishi112FanMask;
+      remote_state[kMitsubishi112FanByte] |= speed;
+      break;
+    default:
+      setFan(kMitsubishi112FanMax);
+  }
+}
+
+// Return the requested state of the unit's fan.
+uint8_t IRMitsubishi112::getFan(void) {
+  return (remote_state[kMitsubishi112FanByte] & kMitsubishi112FanGetMask);
+}
+
+// Return the requested climate operation mode of the a/c unit.
+uint8_t IRMitsubishi112::getMode(void) {
+  return remote_state[kMitsubishi112ModeByte] & kMitsubishi112ModeMask;
+}
+
+// Set the requested climate operation mode of the a/c unit.
+void IRMitsubishi112::setMode(const uint8_t mode) {
+  // If we get an unexpected mode, default to AUTO.
+  switch (mode) {
+    case kMitsubishi112Cool:
+    case kMitsubishi112Heat:
+    case kMitsubishi112Auto:
+    case kMitsubishi112Dry:
```

Comment:
Does this device not have a "Fan" only operation mode? The MITSUBISHI136 was an odd device that didn't have it, just concerned you might have accidentally copied it without knowing knowing that.

Reply:
No it doesn't according to the manual. The MSH model has heat instead of fan only, whereas the MS model, has "Fan" only rather than the heat setting.

Reply:
Great. Thanks for confirming. I just wanted to be sure.